sky/serve/replica_managers.py

```diff
@@ -418,7 +418,13 @@ def url(self) -> Optional[str]:
         handle = self.handle()
         if handle is None:
             return None
-        return f'{handle.head_ip}:{self.replica_port}'
+        try:
+            endpoint = backend_utils.get_endpoints(handle.cluster_name,
+                                                   self.replica_port)
+            assert isinstance(endpoint, str)
+            return endpoint
+        except RuntimeError:
+            return None
```

Comment:
How about we add `http://` in this function directly, instead of handling it in callers?